include/merlin/core_kernels.cuh

```diff
@@ -793,6 +793,35 @@ __forceinline__ __device__ OverrideResult try_override_min_meta(
   return OverrideResult::CONTINUE;
 }
 
+template <class K, class V, class M>
+__forceinline__ __device__ OverrideResult try_swap_min_meta(
+    Bucket<K, V, M>* __restrict bucket, const int key_pos, M min_meta,
+    const M* __restrict metas, M* __restrict evicted_metas, const int key_idx) {
+  if (metas == nullptr) {
+    const M cur_meta = bucket->cur_meta.load(cuda::std::memory_order_relaxed);
+    if (bucket->metas[key_pos].compare_exchange_strong(
+            min_meta, cur_meta + 1, cuda::std::memory_order_relaxed)) {
+      evicted_metas[key_idx] =
```

Comment:
I believe after the `compare_exchange_strong` calling, the `cur_meta` maybe change by other threads, so I think "evicted_metas[key_idx] = cur_meta;" should be what you want, is it right?

Reply:
You need store a snapshot not the newest value of cur_meta.

Reply:
You are right! Value in min_meta has been swapped from bucket->metas[key_pos], if exchange return true. I changed it to be `evicted_metas[key_ids] = min_meta`.